system/jlib/jutil.cpp

```diff
@@ -3056,19 +3175,481 @@ extern jlib_decl void setSecretMount(const char * path)
         secretDirectory.set(path);
 }
 
-extern jlib_decl StringBuffer & getSecret(StringBuffer & result, const char * name, const char * key)
+enum class CVaultKind { kv_v1, kv_v2 };
```

Comment:
is it worth splitting this code into a jsecret.hpp/cpp?  (I don't have a strong opinion either way.)

Reply:
Yes I'll split it out.  Wanted an initial sanity check on the approach first.